app/models/validations/sales/setup_validations.rb

```diff
@@ -0,0 +1,9 @@
+module Validations::Sales::SetupValidations
+  def validate_saledate(record)
+    return unless record.saledate
+
+    if record.saledate < Time.zone.local(2022, 4, 1)
+      record.errors.add :saledate, I18n.t("validations.setup.saledate.financial_year")
+    end
+  end
+end
```

Comment:
would we not need to test also that the saledate is before (2023, 4, 1)
I wonder if ruby comes with some exciting date range inclusion methods

edit: it does
```
[12] pry(main)> new_year = Date.new(2023, 1, 1)
=> #<Date: 2023-01-01 ((2459946j,0s,0n),+0s,2299161j)>
[13] pry(main)> last_week = Date.new(2023, 1, 20)
=> #<Date: 2023-01-20 ((2459965j,0s,0n),+0s,2299161j)>
[14] pry(main)> tomorrow = Date.new(2023, 1, 30)
=> #<Date: 2023-01-30 ((2459975j,0s,0n),+0s,2299161j)>
[15] pry(main)> Date.today.between?(last_week, tomorrow)
=> true
[16] pry(main)> Date.today.between?(new_year, last_week)
=> false
```